src/main/java/net/minecraftforge/datafix/ForgeDataFixDelegate.java

```diff
@@ -0,0 +1,102 @@
+package net.minecraftforge.datafix;
+
+import com.mojang.datafixers.DataFix;
+import com.mojang.datafixers.TypeRewriteRule;
+import net.minecraftforge.fml.util.ObfuscationReflectionHelper;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * This delegate allows for rebuilding of the cached rule, which is needed since it is
+ */
+class ForgeDataFixDelegate extends DataFix
+{
+
+    private static final Method makeRuleMethod = ObfuscationReflectionHelper.findMethod(
+      DataFix.class,
+      "makeRule"
+    );
```

Comment:
So I though so to, but it turns out, no.
You can not simply extend the base class and then access a protected method from a different instance, which is potentially not in your inheritance tree.

This is only possible if the targeted object is guaranteed to be in your inheritance tree, which is not the case here. Both Lex and I thought too that it was possible, but the compiler just does not allow it.

Reply:
And yes we use ORH here to prevent the reflection gore and provide better future compatibility with regards to reflection and its limitations in the newer JVM version.